test/higher_order_ops/test_with_effects.py

```diff
@@ -662,6 +680,241 @@ def forward(self, arg0_1, arg1_1):
     return (getitem, mul, mul)""",
         )
 
+    @skipIfNoDynamoSupport
+    def test_effectful_op_in_backward(self):
+        with torch.library._scoped_library("_mylib", "FRAGMENT") as lib:
+            lib.define("foo(Tensor x) -> Tensor")
+
+            def foo_impl(a):
+                return a.clone()
+
+            def foo_bwd(ctx, grad):
+                return torch.ops._mylib.foo(grad)
+
+            for backend in ["CPU", "CUDA", "Meta"]:
+                lib.impl("foo", foo_impl, backend)
+
+            if torch._C._dispatch_has_kernel_for_dispatch_key(
+                "_mylib::foo", "Autograd"
+            ):
+                self.skipTest(
+                    "Double registration of Autograd kernel for test custom op"
+                )
```

Comment:
We shouldn't need this code

Reply:
Yes, it was failing only on CI in parallel execution, wanted to debug this later. I think lib=lib should fix this, will add it.